packages/explat-client/src/internal/requests.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * Internal dependencies
+ */
+import { Config, ExperimentAssignment } from '../types';
+import { validateExperimentAssignment } from './validations';
+import { monotonicNow } from './timing';
+import { isObject } from './validations';
+import * as ExperimentAssignments from './experiment-assignments';
+
+interface FetchExperimentAssignmentResponse {
+	variations: Record< string, unknown >;
+	ttl: number;
+}
+
+function isFetchExperimentAssignmentResponse(
+	response: unknown
+): response is FetchExperimentAssignmentResponse {
+	return (
+		isObject( response ) && isObject( response.variations ) && typeof response.ttl === 'number'
+	);
+}
+
+function validateFetchExperimentAssignmentResponse(
+	response: unknown
+): FetchExperimentAssignmentResponse {
+	if ( isFetchExperimentAssignmentResponse( response ) ) {
+		return response;
+	}
+	throw new Error( 'Invalide FetchExperimentAssignmentResponse' );
+}
+
+export async function fetchExperimentAssignment(
+	config: Config,
+	experimentName: string
+): Promise< ExperimentAssignment > {
+	const retrievedTimestamp = monotonicNow();
+
+	const { variations, ttl } = validateFetchExperimentAssignmentResponse(
+		await config.fetchExperimentAssignment( {
+			anonId: ( await config.getAnonId() ) ?? undefined,
```

Comment:
I think this is the only place it will be used, and I'd like to be explicit here that we want an undefined if we don't have an `anonId`.

I tend to prefer `null` over `undefined` for explicitly missing data as opposed to implicit.

Reply:
We also check for it in the body of the `config.fetchExperimentAssignment()` implementation we pass in. So we have something like:

* `getAnonId()` may return `null` or `string`
* `config.fetchExperimentAssignment()` expects the `anonId` param to be `string` or `undefined`

So if `getAnonId()` returns `null`, we pass it into `config.fetchExperimentAssignment()` as `undefined`, which also has the `anonId ?? undefined` check.

I see your point about being explicit about missing data, but maybe in this case, we should define `anonId` as `string | null` in `config.fetchExperimentAssignment()`?